### PR TITLE
Moves the GPS component from the upload computers and communications console to the boards themselves when used

### PR DESCRIPTION
**Pull request by StrangeWeirdKitten**
## About The Pull Request

Moves the GPS component on Upload computers and Communications Console to be handled from the board, not the machine, when a board is used to modify the AI or used to login.

## Why It's Good For The Game
Quite frankely, the mechanic already exists. But it does not work due to the component being attatched to the computer.

Currently it's intended for these critical computers to give GPS signals because they can be taken, and built extremely easily, anywhere, and deconstructed. Especially the upload boards, since the AI and borgs are directly in the center of it when people keep subverting and fixing laws. This is extremely annoying for the silicons. On top of that, you building a communications console to hack in maint somewhere will leave some sort of a trace to justify the reward.

Yes, they have a computer that sends out the GPS signal. But anyone with a brain could realize all that's needed to circumvent this mechanic is the use of a screwdriver.

Now, the GPS is registered on the boards themselves. The boards get stored in the contents of the computer. You can't circumvent this now. 

## Changelog

:cl:
balance: The upload and communication boards directly have trackers installed, activated only when authenticated.
/:cl:

### CODE REVIEW

**Comment by YuliceTheDruid** [comment]
Me when I print a bunch of boards and throw them all around maint and then build one in particular.

**Comment by StrangeWeirdKitten** [comment]
> Me when I print a bunch of boards and throw them all around maint and then build one in particular.

Sure if you want to go this effort in subverting the AI, you deserve to subvert it. Make sure you scrub each board with soap though

**Comment by YuliceTheDruid** [comment]
> Sure if you want to go this effort in subverting the AI, you deserve to subvert it. Make sure you scrub each board with soap though

I will shrimply wear black gloves and every seccie will be confused.

**Comment by ZephyrTFA** [comment]
Given these can be printed en masse with zero restrictions this isn't gonna happen

**Comment by ZephyrTFA** [comment]
agreement reached in discord, pending code

**Comment by StrangeWeirdKitten** [comment]
I just looked at the communications console, I'm not touching that code. The PR is only going to tackle the more pressing uploads.

**Comment by StrangeWeirdKitten** [comment]
I'll make it a proc for machinery/computer at another point when I feel like properly addressing the communications console. For now I'm just primarily looking at the more important duo.

**Review by ZephyrTFA — APPROVED** [review]

**Comment by Jacquerel** [comment]
As a very minor thing could we add some kind of feedback note (perhaps also when examining the board, probably also when it is flipped on for the first time) informing someone that the item they are holding has its GPS activated?

**Comment by StrangeWeirdKitten** [comment]
> As a very minor thing could we add some kind of feedback note (perhaps also when examining the board, probably also when it is flipped on for the first time) informing someone that the item they are holding has its GPS activated?

Done

I also just hooked the proc to run inside communication console login and expanded it to be used on any computer in the future.

File: code/game/objects/items/circuitboards/computer_circuitboards.dm [hunk 1]
```diff
@@ -2,6 +2,11 @@
 	name = "Generic"
 	name_extension = "(Computer Board)"
 
+/obj/item/circuitboard/computer/examine()
+	if(GetComponent(/datum/component/gps))
+		. = ..()
+		. += span_info("there's a small, blinking light!")
```
Comment (Jacquerel) [inline code/game/objects/items/circuitboards/computer_circuitboards.dm:8]:
```suggestion
	. = ..()
	if(GetComponent(/datum/component/gps))
		. += span_info("there's a small, blinking light!")
```

File: code/game/machinery/computer/_computer.dm [hunk 2]
```diff
@@ -87,6 +87,16 @@
...
 
+/obj/machinery/computer/proc/imprint_gps(gps_tag) // Currently used by the upload computers and communications console
+	var/tracker = gps_tag
+	if(!tracker) // Don't give a null GPS signal if there is none
+		return
+	for(var/obj/item/circuitboard/computer/board in src.contents)
+		if(!contents || board.GetComponent(/datum/component/gps))
+			return
+		board.AddComponent(/datum/component/gps, "[tracker]")
+		balloon_alert_to_viewers("Board Authenticated and Imprinted", vision_distance = 1)
```
Comment (Jacquerel) [inline code/game/machinery/computer/_computer.dm:98]:
```suggestion
		balloon_alert_to_viewers("board tracker enabled", vision_distance = 1)
```
or `board authenticated` or `board imprinted` or whatever else.
Convention with balloon alerts is that they should be entirely lower case, and also as short as reasonably possible.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Comment by StrangeWeirdKitten** [comment]
👍

**Pull request merged**